st7735r.h

```diff
@@ -21,6 +21,9 @@
 #elif defined(ST7735R)
```

Comment:
Maybe change this to

```c
#if defined(PIMORONI_DISPLAY_HAT_MINI)
#define DISPLAY_NATIVE_WIDTH 240
#define DISPLAY_NATIVE_HEIGHT 320
#elif ...
...
#endif
```
and throughout the codebase, reuse the define `ST7789` to drive the display to avoid introducing `ST7789V2` and `ST7789` that basically mean the same controller (but with 240x240 vs 240x320 difference).

The "native width/height" concept is intended to denote the native scanline order of the panel. The native scan order is the direction that the display updates pixels, which I believe is always portrait orientation in all these display or 240x320 pixels in size. I never met a display panel that would have a native scan order of 320x240 pixels. So the appropriate line here would be
```c
#define DISPLAY_NATIVE_WIDTH 240
#define DISPLAY_NATIVE_HEIGHT 320
```
The native scan order should be the same as the native memory upload/read interpretation order when the MADCTL register is set to all zero bits.

Reply:
Oh hmm actually, reading https://pinout.xyz/pinout/display_hat_mini# there is a mention of "ST7789V2" as if it is a separate controller name. Are there any markings/names on the display hat board itself?

Reply:
I have a strong suspicion that the native scan order of the ST7789V2 is actually 320x240, but I would have to run some tests to verify this.

Raspberry Pi OS Bullseye has just dropped a whole world of pain into my lap, though, so it may be some time before I do.